rocky/reports/report_types/mail_report/report.py

```diff
@@ -0,0 +1,70 @@
+from datetime import datetime
+from logging import getLogger
+from typing import Any, Dict, List
+
+from django.utils.translation import gettext_lazy as _
+
+from octopoes.models import Reference
+from octopoes.models.ooi.dns.zone import Hostname
+from octopoes.models.ooi.network import IPAddressV4, IPAddressV6
+from reports.report_types.definitions import Report
+
+logger = getLogger(__name__)
+
+
+class MailReport(Report):
+    id = "mail-report"
+    name = _("Mail Report")
+    description = _("System specific mail report that focusses on IP addresses and hostnames.")
+    plugins = {"required": ["nmap"], "optional": []}
+    input_ooi_types = {Hostname, IPAddressV4, IPAddressV6}
+    template_path = "mail_report/report.html"
+
+    def generate_data(self, input_ooi: str, valid_time: datetime) -> Dict[str, Any]:
+        reference = Reference.from_str(input_ooi)
+        hostnames = []
+        mail_security_measures = {}
+
+        if reference.class_type == Hostname:
+            hostnames = [reference]
+            measures = self._get_measures(valid_time, hostnames)
+            mail_security_measures = {"hostname": hostnames[0].tokenized.name, "measures": measures}
+        elif reference.class_type in (IPAddressV4, IPAddressV6):
+            hostnames = self.octopoes_api_connector.query(
+                "IPAddress.<address[is ResolvedHostname].hostname", valid_time, reference
+            )
+            for hostname in hostnames:
```

Comment:
At this point we have no guarantee we are actually checking mailservers, so I think we need to add a similar check from the system report

Reply:
See:
https://github.com/minvws/nl-kat-coordination/blob/acd82367afbd7b0285389f1ce2e672a081265b51/rocky/reports/report_types/systems_report/report.py#L49-L84

Here we look for smtp-like services as defined on line 54 to 58

Reply:
Lets chat about this! In aggregate report we check this with the results of systems report. Seems like double work.

Reply:
Yes ok so for IP addresses we don't need those checks if we let the user decide if it needs a mail report :ok_hand: